CyvoreOS/phone_numbers_extension.py

```diff
@@ -0,0 +1,270 @@
+    else:
+        logging.warning("The number is not possible.")
+        return "Error - after is_possible_number"
+
+
+def retry_with_country_codes(phone_number: str) -> List[str]:
+    """Function will return a list of the same number with different country codes that validate successfully.
+    Special handling is applied for countries that retain leading zeros in phone numbers.
+    """
+    logging.info("Starting retry with country codes for number: %s", phone_number)
+    valid_numbers = []
+    stripped_from_plus = phone_number.lstrip("+")
+
+    for code in country_codes:
+        try:
+            if code not in special_retain_zero_country_codes:
+                full_number = code + stripped_from_plus.lstrip("0")
+                logging.debug("Trying with country code %s: %s", code, full_number)
+            else:
+                full_number = code + stripped_from_plus  # keep '0'
+                logging.debug(
+                    "< special_retain_zero > Trying country code %s with leading '0' for number: %s",
+                    code,
+                    full_number,
+                )  # ["+39", "+33", "+46"]
+            parsed_number = phonenumbers.parse(full_number)
+            result = validate_number(parsed_number)
+
+            if "Error" not in result and result not in valid_numbers:
+                logging.info(
+                    "Successfully validated number with country code %s: %s",
+                    code,
+                    result,
+                )
+                valid_numbers.append(result)
+
+        except phonenumbers.NumberParseException:
+            logging.warning(
+                "Failed to parse number with country code %s: %s", code, phone_number
+            )
+        except Exception as e:
+            logging.error(
+                "Unexpected error with number %s and country code %s: %s",
+                phone_number,
+                code,
+                e,
+            )
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    if valid_numbers:
+        logging.info(
+            "Retry successful. Total valid numbers found: %d", len(valid_numbers)
+        )
+    else:
+        logging.warning(
+            "Retry failed for all country codes for number: %s", phone_number
+        )
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    # Return the list of valid numbers
+    return valid_numbers
+
+
+# ------------------------------------------------------------------------------------------------------
+
+
+def process_phone_numbers(
+    normalized: List[str],
+) -> List[str]:
+    """Main function to parse and validate phone numbers."""
+    logging.info(
+        "Starting phone number processing. Total normalized numbers to process: %d",
+        len(normalized),
+    )
+    results = set()
+
+    for num in normalized:
+        try:
+            logging.debug("Attempting to parse number: %s", num)
+            parsed = phonenumbers.parse(num)
```

Comment:
When using third party function with unclear name it will be great to add a comment about it, for example what is parsed means here?